lib/srv/sess.go

```diff
@@ -379,15 +379,71 @@ func (s *SessionRegistry) isApprovedFileTransfer(scx *ServerContext) (bool, erro
 		return false, trace.AccessDenied("Teleport user does not match original requester")
 	}
 
-	var incomingShellCmd string
-	if scx.sshRequest != nil {
-		incomingShellCmd = string(scx.sshRequest.Payload)
+	return sess.checkIfFileTransferApproved(req)
+}
+
+type FileTransferRequestEvent string
+
+const (
+	FileTransferUpdate   FileTransferRequestEvent = "file_transfer_request"
+	FileTransferApproved FileTransferRequestEvent = "file_transfer_request_approve"
+	FileTransferDenied   FileTransferRequestEvent = "file_transfer_request_deny"
+)
+
+// NotifyFileTransferRequest is called to notify all members of a party that a file transfer request has been created/approved/denied.
+// The notification is a global ssh request and requires the client to update it's UI state accordingly.
+func (s *SessionRegistry) NotifyFileTransferRequest(req *fileTransferRequest, res FileTransferRequestEvent, scx *ServerContext) error {
+	session := scx.getSession()
+	if session == nil {
+		s.log.Debugf("Unable to notify %s, no session found in context.", res)
+		return nil
```

Comment:
Should this be an error or is it expected that there will sometimes not be a session?

Reply:
Should be an error, thanks!